product_filters/models/product.py

```diff
@@ -74,54 +74,56 @@ def _get_async_ranges(self, category):
         ranges_obj = self.env['product.price.ranges'].search([])
         count_dict = {}
         prod_ids = []
-        if category:
-            prod_ids = prod_obj.search(
-                [('public_categ_ids', 'child_of', int(category)),
-                 ('website_published', '=', True)])
-        if prod_ids:
-            for prod in prod_ids:
-                for ran in ranges_obj:
-                    if ran.upper > prod.list_price > ran.lower:
-                        if ran.id in count_dict.keys():
-                            count_dict[ran.id] += 1
-                        else:
-                            count_dict[ran.id] = 1
-                    if ran.id not in count_dict.keys():
-                        count_dict[ran.id] = 0
-            to_jsonfy = [{'id': k, 'qty': count_dict[k]} for k in count_dict]
-            return to_jsonfy
+        if not category:
+            return []
+        prod_ids = prod_obj.search(
+            [('public_categ_ids', 'child_of', int(category)),
+             ('website_published', '=', True)])
+        if not prod_ids:
+            return []
+        for prod in prod_ids:
+            for ran in ranges_obj:
+                if ran.upper > prod.list_price > ran.lower:
+                    if ran.id in count_dict.keys():
```

Comment:
Similar situation of `dict.set_default`
https://github.com/Vauxoo/ecommerce/pull/4/files#r70112681